### PR TITLE
Improve device init

### PR DESCRIPTION
**Pull request by jschlyter**
Clean up device init
Use VIN as device identifier (not name) as this never change

### CODE REVIEW

File: custom_components/polestar_api/sensor.py [hunk 1]
```diff
@@ -542,10 +541,6 @@ def __init__(
         if self.device is not None and self.device.get_latest_call_code() == 200:
             self._async_update_attrs()
 
-    def _get_current_value(self) -> StateType | None:
```
Comment (jschlyter) [inline custom_components/polestar_api/sensor.py:545]: Unused code removed

File: custom_components/polestar_api/polestar.py [hunk 2]
```diff
@@ -17,40 +19,55 @@
...
-        return self.vin[-4:]
-
-    async def init(self):
+    async def init(self, car: int = 0):
         """Initialize the Polestar API."""
         await self.polestarApi.init()
+        self.car = car
+        self.polestarApi.set_car_data(car)
 
-    def set_car_data(self, index: int = 0):
```
Comment (jschlyter) [inline custom_components/polestar_api/polestar.py:42]: Now part if init()

File: custom_components/polestar_api/polestar.py [hunk 3]
```diff
@@ -17,40 +19,55 @@
...
-        """Set the car data."""
-        self.polestarApi.set_car_data(index)
-
-    def set_vin(self):
```
Comment (jschlyter) [inline custom_components/polestar_api/polestar.py:46]: Now part if init()

File: custom_components/polestar_api/__init__.py [hunk 4]
```diff
@@ -39,11 +39,9 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
 
         # for each number of car we are going to create a new entry
         entities: list[Polestar] = []
-        for index in range(number_of_cars):
+        for car_index in range(number_of_cars):
             polestar = Polestar(hass, conf[CONF_USERNAME], conf[CONF_PASSWORD])
-            await polestar.init()
```
Comment (jschlyter) [inline custom_components/polestar_api/__init__.py:44]: Everything set up in `init` (should perhaps be renamed to `async_init` ?

**Review by leeyuentuen — APPROVED** [review]

**Review by leeyuentuen — APPROVED** [review]

**Pull request merged**